api/v1alpha1/system_types.go

```diff
@@ -874,6 +950,116 @@ func (tc *ThinkingSpec) Default() {
+type SystemSearchdSpec struct {
+	// Deploy searchd instance
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Enabled *bool `json:"enabled,omitempty"`
+	// Image specification for the Searchd component.
+	// Defaults to system image if not defined.
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Image *ImageSpec `json:"image,omitempty"`
+	// Configuration options for System's Searchd
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Config *SearchdConfig `json:"config,omitempty"`
+	// Resource requirements for the Searchd component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Resources *ResourceRequirementsSpec `json:"resources,omitempty"`
+	// Liveness probe for the Searchd component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	LivenessProbe *ProbeSpec `json:"livenessProbe,omitempty"`
+	// Readiness probe for the Searchd component
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	ReadinessProbe *ProbeSpec `json:"readinessProbe,omitempty"`
+	// Describes node affinity scheduling rules for the Searchd pod
+	// +optional
+	NodeAffinity *corev1.NodeAffinity `json:"nodeAffinity,omitempty" protobuf:"bytes,1,opt,name=nodeAffinity"`
+	// If specified, the Searchd pod's tolerations.
+	// +optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty" protobuf:"bytes,22,opt,name=tolerations"`
+	// Configures the TerminationGracePeriodSeconds for Searchd
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	TerminationGracePeriodSeconds *int64 `json:"terminationGracePeriodSeconds,omitempty"`
+}
+
+// Default implements defaulting for the system sphinx component
+func (spec *SystemSearchdSpec) Default() {
+	spec.Enabled = boolOrDefault(spec.Enabled, pointer.Bool(systemDefaultSearchdEnabled))
+	spec.Image = InitializeImageSpec(spec.Image, defaultImageSpec(systemDefaultSearchdImage))
+	spec.Resources = InitializeResourceRequirementsSpec(spec.Resources, systemDefaultSearchdResources)
+	spec.LivenessProbe = InitializeProbeSpec(spec.LivenessProbe, systemDefaultSearchdLivenessProbe)
+	spec.ReadinessProbe = InitializeProbeSpec(spec.ReadinessProbe, systemDefaultSearchdReadinessProbe)
+	if spec.Config == nil {
+		spec.Config = &SearchdConfig{}
+	}
+	spec.Config.Default()
+	spec.TerminationGracePeriodSeconds = int64OrDefault(
+		spec.TerminationGracePeriodSeconds, systemDefaultTerminationGracePeriodSeconds,
+	)
+}
+
+// SearchdConfig has configuration options for System's sphinx
+type SearchdConfig struct {
+	// Allows setting the service name for Searchd
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	ServiceName *string `json:"serviceName,omitempty"`
+	// The TCP port Searchd will run its daemon on
+	// +operator-sdk:csv:customresourcedefinitions:type=spec
+	// +optional
+	Port *int32 `json:"port,omitempty"`
```

Comment:
I would say that new search image does not support most (or none) of current envvars like `BindAddress`, `Port`, `ConfigFile`, `BatchSize`, `PIDFile`... I think only the path and PVC config is supported

Checking statefulset config (`search_statefulset.go` in the generator), there are 0 ENV vars, so if not used, why are there referenced on the new search statefulset CR config?

Reply:
You are right, probably they won't be used at all.

Reply:
You need also to remove them from the default vars in that file

Reply:
Fixed in https://github.com/3scale-ops/saas-operator/pull/259/commits/3a6eefbd918b523795b2ed9f1687f02a340daaf3, among several other cleanups included in [0.19.7-alpha.6](https://github.com/3scale-ops/saas-operator/pull/259/commits/72fbe6681bac85fb9d877e1fca5d4221d487fb10)